### PR TITLE
Add coercion for union

### LINKED ISSUE #2532: Coercion doesn't work for UNION
select * from (values (1)) union all (values (NULL));

fails with:

Query 20150316_192120_21072_kzg7u failed: column 0 in union query has incompatible types: bigint, unknown

### PR DESCRIPTION
**Pull request by haozhun**
Fixes #2532

### CODE REVIEW

File: presto-parser/src/main/java/com/facebook/presto/sql/SqlFormatter.java [hunk 1]
```diff
@@ -360,6 +360,7 @@ protected Void visitValues(Values node, Integer indent)
                 builder.append(formatExpression(row));
                 first = false;
             }
+            builder.append("\n");
```
Comment (dain) [inline presto-parser/src/main/java/com/facebook/presto/sql/SqlFormatter.java]: Is this an unrelated change?

Comment (haozhun) [inline presto-parser/src/main/java/com/facebook/presto/sql/SqlFormatter.java]: Moved it to a separate commit.

File: presto-main/src/main/java/com/facebook/presto/sql/analyzer/TupleAnalyzer.java [hunk 2]
```diff
@@ -371,12 +371,11 @@ protected TupleDescriptor visitUnion(Union node, AnalysisContext context)
 
         TupleAnalyzer analyzer = new TupleAnalyzer(analysis, session, metadata, sqlParser, experimentalSyntaxEnabled);
 
-        // Use the first descriptor as the output descriptor for the UNION
-        TupleDescriptor outputDescriptor = analyzer.process(node.getRelations().get(0), context).withOnlyVisibleFields();
+        TupleDescriptor[] descriptors = node.getRelations().stream().map(r -> analyzer.process(r, context).withOnlyVisibleFields()).toArray(TupleDescriptor[]::new);
```
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/sql/analyzer/TupleAnalyzer.java]: `relation` instead of `r` for the variable

File: presto-main/src/main/java/com/facebook/presto/sql/analyzer/TupleAnalyzer.java [hunk 3]
```diff
@@ -371,12 +371,11 @@ protected TupleDescriptor visitUnion(Union node, AnalysisContext context)
...
 
-        for (Relation relation : Iterables.skip(node.getRelations(), 1)) {
-            TupleDescriptor descriptor = analyzer.process(relation, context).withOnlyVisibleFields();
-            int outputFieldSize = outputDescriptor.getVisibleFields().size();
+        Type[] outputFieldTypes = descriptors[0].getVisibleFields().stream().map(f -> f.getType()).toArray(Type[]::new);
```
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/sql/analyzer/TupleAnalyzer.java]: `field` instead of `f`

File: presto-main/src/main/java/com/facebook/presto/sql/planner/RelationPlanner.java [hunk 4]
```diff
@@ -573,6 +574,42 @@ public Expression rewriteExpression(Expression node, Void context, ExpressionTre
         }
     }
 
+    private RelationPlan processAndCoerceIfNecessary(Relation node, Void context)
+    {
+        Type[] coerceToTypes = analysis.getRelationCoercion(node);
+
+        RelationPlan plan = node.accept(this, context);
+
+        if (coerceToTypes != null) {
```
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/RelationPlanner.java]: this will be easier to read if you invert this to 
```
if (coerceToTypes == null) {
  return plan;
}
...
```

**Comment by cberner** [comment]
Looks good
